### PR TITLE
Added static-DHCP feature

### PR DESCRIPTION
**Pull request by JunqiZhang0**

### CODE REVIEW

**Comment by abraverm** [comment]
Please provide and an example, update documentation and fix tests.

File: linchpin/provision/roles/libvirt/files/schema.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
                     "vcpus": { "type": "integer", "required": true },
                     "cpu_mode": { "type": "string", "required": false },
                     "memory": { "type": "integer", "required": true },
+                    "static_ip": {"type": "boolean","required": false },
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/files/schema.json]: I think this should be a network option: VM might be connected on many networks, but only for some its important to have a static dhcp

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 2]
```diff
@@ -410,6 +410,15 @@
   delay: 10
   when: uri_hostname == 'localhost'
 
+- set_fact: 
+    static_ip: "{{extract_ip_address_result.results.0['stdout']|omit_filter(omit)}}"
+  when: uri_hostname == 'localhost'
+
+- debug:
+    msg: "{{ macs }}"
+# - debug:
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: remove commented code

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 3]
```diff
@@ -410,6 +410,15 @@
   delay: 10
   when: uri_hostname == 'localhost'
 
+- set_fact: 
+    static_ip: "{{extract_ip_address_result.results.0['stdout']|omit_filter(omit)}}"
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: How do you know its the right IP? or later in 485 which network to update?

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 4]
```diff
@@ -423,6 +432,12 @@
   delay: 10
   remote_user: "{{ res_def['remote_user'] | default(ansible_user_id) }}"
   delegate_to: "{{ uri_hostname }}"
+  # set_fact: 
+  #   static_ip: "{{extract_ip_address_result_remote.results.0['stdoutq']|omit_filter(omit)}}"
+  when: uri_hostname != 'localhost'
+
+- set_fact: 
+    static_ip: "{{extract_ip_address_result_remote.results.0['stdout']|omit_filter(omit)}}"
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: duplicate code of 413

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 5]
```diff
@@ -410,6 +410,15 @@
   delay: 10
   when: uri_hostname == 'localhost'
 
+- set_fact: 
+    static_ip: "{{extract_ip_address_result.results.0['stdout']|omit_filter(omit)}}"
+  when: uri_hostname == 'localhost'
+
+- debug:
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: Does the debug info will help the user in someway?

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 6]
```diff
@@ -463,3 +478,15 @@
   when:
     - extract_ip_address_result is not defined
     - extract_ip_address_result_remote is not defined
+- debug: 
+    msg: "{{ res_def['name'] }}"
+  
+- name: "Update the ip in Inventory"
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: How about using https://docs.ansible.com/ansible/latest/modules/virt_net_module.html ?

Comment (samvarankashyap) [reply linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: @JunqiZhang0 
I agree with @abraverm . virt_net module might help

Comment (JunqiZhang0) [reply linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: Ok I'll change it

**Comment by samvarankashyap** [comment]
@JunqiZhang0 is this duplicate of #1198 if so please close it.

**Comment by JunqiZhang0** [comment]
@closed

**Pull request closed without merging**